src/components/CategoriesBar.jsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from "react";
+import * as api from "./api";
+import { useNavigate } from "react-router-dom";
+
+export const CategoriesBar = ({ setDisplayCategories }) => {
+    const [categoryApi, setCategoryApi] = useState([]);
+    const [selectCategory, setSelectCategory] = useState("");
```

Comment:
It's best to initialise your states to the same datatype you use throughout your code - for example later you're storing `selectCategory` in an object, so I'd initialise it as an object rather than a string